src/parser/SpatialQuery.cpp

```diff
@@ -0,0 +1,149 @@
+// Copyright 2024, University of Freiburg
+// Chair of Algorithms and Data Structures
+// Authors: Christoph Ullinger <ullingec@informatik.uni-freiburg.de>
+
+#include "parser/SpatialQuery.h"
+
+namespace parsedQuery {
+
+// ____________________________________________________________________________
+void SpatialQuery::addParameter(const SparqlTriple& triple) {
+  auto simpleTriple = triple.getSimple();
+  TripleComponent predicate = simpleTriple.p_;
+  TripleComponent object = simpleTriple.o_;
+
+  if (!predicate.isIri()) {
+    throw SpatialSearchException("Predicates must be IRIs");
+  }
+
+  std::string predString = predicate.getIri().toStringRepresentation();
+  if (predString.ends_with("left>")) {
+    setVariable("left", object, left_);
+  } else if (predString.ends_with("right>")) {
+    setVariable("right", object, right_);
+  } else if (predString.ends_with("nearestNeighbors>")) {
```

Comment:
I don't like the `ends_with` , please check for the full predicates.

Reply:
multiple occurences also below.

Reply:
I agree, that this is not very clean. I will change it here and also in the Path Search code from where this was inspired. The main advantage is that you can use it with or without the spatialSearch: predicate, but this should rather be achieved through an "or"...